### ISSUE TITLE
Transform class needs better implementation

### ISSUE THREAD

**Issue by raidho36** [raidho36/love-multitouch#1]
Current implementation uses step by step vector transformation algorithm, and that takes 8 multiplications and 10 additions. I haven't been able to reliably solve this using matrices: while forward transformation worked fine, inverse transformation (the important one) would fail to get coordinates back to its original position. My best effort ended with it being able to do this but only as long as scaling was uniform or there was no shearing. I would add a bit more code to properly compensate for said inaccuracy but that wound up being just as expensive as step by step transformation while using up more variables.

**Comment by MartyMaro** [comment]
Has this issue been addressed?

**Comment by raidho36** [comment]
No. I haven't been working on the library, mostly because it felt complete
already.

2018-02-09 4:45 GMT+03:00 CodingMog <notifications@github.com>:
